fastavro/writer.py

```diff
@@ -425,7 +425,11 @@ def dump():
     acquaint_schema(schema)
 
     for record in records:
-        write_data(io, record, schema)
+        try:
```

Comment:
You are correct that the original exception would get swallowed. This was done intentionally since I thought the exception from ```validate_data``` would be superior, but thinking about it again I think it's good to keep the original exception, so that can be fixed.

As for the benchmarking, I'll try to do some and post the results. From what I've read, try/excepts that don't raise an exception are extremely cheap, but I'll see if I can get some measurements.